packages/components/src/color-picker/use-deprecated-props.ts

```diff
@@ -69,7 +69,10 @@ function isLegacyProps( props: any ): props is LegacyProps {
 
 function getColorFromLegacyProps(
 	props: LegacyProps
-): ColorFormats.HSL | ColorFormats.HSLA {
+): ColorFormats.HSL | ColorFormats.HSLA | undefined {
+	if ( ! props.color ) {
```

Comment:
Looking at the `LegacyProps` type, `props.color` should be always defined.

What's the case for `color` potentially not being defined? In that case, we should update the `LegacyProps` type